solidity/test/helpers/initContracts.js

```diff
@@ -36,36 +38,45 @@ async function initContracts(KeepToken, TokenStaking, KeepRandomBeaconService,
           registry.address,
       ).encodeABI();
 
-  serviceContractProxy = await KeepRandomBeaconService.new(serviceContractImplV1.address, initialize);
-  serviceContract = await KeepRandomBeaconServiceImplV1.at(serviceContractProxy.address);
+  serviceContractProxy = await KeepRandomBeaconService.new(serviceContractImplV1.address, initialize, {from: accounts[0]});
 
+  serviceContract = await KeepRandomBeaconServiceImplV1.at(serviceContractProxy.address);
   // Initialize Keep Random Beacon operator contract
-  const bls = await BLS.new();
+  const bls = await BLS.new({from: accounts[0]});
+  await KeepRandomBeaconOperator.detectNetwork()
   await KeepRandomBeaconOperator.link("BLS", bls.address);
-  const groupSelection = await GroupSelection.new();
-  const groups = await Groups.new();
-  const dkgResultVerification = await DKGResultVerification.new();
-  const reimbursements = await Reimbursements.new();
+  const groupSelection = await GroupSelection.new({from: accounts[0]});
+  await Groups.detectNetwork()
+  await Groups.link("BLS", bls.address);
+  const groups = await Groups.new({from: accounts[0]});
+
+  const dkgResultVerification = await DKGResultVerification.new({from: accounts[0]});
+
+  const reimbursements = await Reimbursements.new({from: accounts[0]});
+
+  await KeepRandomBeaconOperator.detectNetwork()
   await KeepRandomBeaconOperator.link("GroupSelection", groupSelection.address);
+  await KeepRandomBeaconOperator.detectNetwork()
   await KeepRandomBeaconOperator.link("Groups", groups.address);
+  await KeepRandomBeaconOperator.detectNetwork()
   await KeepRandomBeaconOperator.link("DKGResultVerification", dkgResultVerification.address);
+  await KeepRandomBeaconOperator.detectNetwork()
```

Comment:
Why do we need to repeat `detectNetwork` after each `.link`?

Reply:
oops looks like bad search and replace